node_shell.py

```diff
@@ -387,10 +851,271 @@ def do_exit(self, arg):  # pylint: disable=unused-argument
+                              action='store_true')
+    group_parser.add_argument('--startswith',
+                              '-s',
+                              help='Filter by the initial string')
+    group_parser.add_argument('--endswith',
+                              '-e',
+                              help='Filter by ending string')
+    group_parser.add_argument(
+        '--contains',
+        '-c',
+        help='Filter by checking if the group name contains the string')
+    group_parser.add_argument(
+        '--past-days',
+        '-p',
+        type=int,
+        help='Filter by only including those created in the past N days')
+    group_parser.add_argument(
+        '--count',
+        '-C',
+        action='store_true',
+        help='Show also the number of nodes in the group')
+
+    @cmd2.with_argparser(group_parser)
+    def do_group_list(self, args):
+        """Command for list groups"""
+        from aiida.cmdline.commands.cmd_group import group_list
+        with self.verdi_isolate():
+            group_list.callback(all_users=args.all_users,
+                                user_email=args.user_email,
+                                all_types=args.all_types,
+                                group_type=args.group_type,
+                                with_description=args.with_description,
+                                count=args.count,
+                                past_days=args.past_days,
+                                startswith=args.startswith,
+                                endswith=args.endswith,
+                                contains=args.contains,
+                                node=None)
+
+    @cmd2.with_argparser(group_parser)
+    @needs_node
+    def do_group_belong(self, args):
+        """Command for list groups"""
+        from aiida.cmdline.commands.cmd_group import group_list
+        with self.verdi_isolate():
+            group_list.callback(all_users=args.all_users,
+                                user_email=args.user_email,
+                                all_types=args.all_types,
+                                group_type=args.group_type,
+                                with_description=args.with_description,
+                                count=args.count,
+                                past_days=args.past_days,
+                                startswith=args.startswith,
+                                endswith=args.endswith,
+                                contains=args.contains,
+                                node=self._current_node)
+
+    def do_cd(self, args):
+        """Change directory"""
+        path = args
+        if not path:
+            path = os.path.expanduser("~")
+        os.chdir(path)
+
+    complete_cd = cmd2.Cmd.path_complete
+
+    @with_default_argparse
+    def do_pwd(self, arg):
+        """Return the current workding directory"""
+        self.poutput(os.getcwd())
+
+    @with_default_argparse
+    def do_vshell(self, _):
```

Comment:
Let's call it `verdi_shell` and then people can do an alias?

Also, we should try to clarify the advantage of running `verdi_shell` vs. `verdi shell` as they look a bit different. I guess it's the fact that one gets the current node, right? Also, maybe one can always execute the `%colors` magic, or there are issues with that?